WordPress/Classes/ViewRelated/Tools/PromptViewController.swift

```diff
@@ -0,0 +1,129 @@
+    /// -   (And verything) inside a UINavigationController instance.
+    ///
+    static func navigationControllerWithPrompt<T: UIViewController where T: Confirmable>(viewController: T) -> UINavigationController {
+        let viewController = PromptViewController(viewController: viewController)
+        return UINavigationController(rootViewController: viewController)
+    }
+    
+    
+    
+    /// MARK: - Initializers / Deinitializers
+    
+    deinit {
+        stopListeningToProperties(childViewController)
+    }
+    
+    init(viewController: UIViewController) {
+        // You stay with us, sir
+        childViewController = viewController
+        
+        super.init(nibName: nil, bundle: nil)
+        precondition(viewController.conformsToProtocol(Confirmable))
+        
+        setupNavigationButtons()
+        attachChildrenViewController(viewController)
+        setupChildrenViewConstraints(viewController.view)
+        startListeningToProperties(viewController)
+    }
+    
+    required init?(coder aDecoder: NSCoder) {
+        fatalError("Not meant for Nib usage!")
+    }
+    
+    
+    
+    // MARK: - Private Helpers
+    
+    private func attachChildrenViewController(viewController: UIViewController) {
+        // Attach!
+        viewController.willMoveToParentViewController(self)
+        view.addSubview(viewController.view)
+        addChildViewController(viewController)
+        viewController.didMoveToParentViewController(self)
+    }
+    
+    private func setupChildrenViewConstraints(childrenView : UIView) {
+        // We grow, you grow. We shrink, you shrink. Capicci?
+        childrenView.translatesAutoresizingMaskIntoConstraints = false
+        childrenView.widthAnchor.constraintEqualToAnchor(view.widthAnchor).active = true
+        childrenView.heightAnchor.constraintEqualToAnchor(view.heightAnchor).active = true
+        childrenView.leftAnchor.constraintEqualToAnchor(view.leftAnchor).active = true
+        childrenView.topAnchor.constraintEqualToAnchor(view.topAnchor).active = true
+    }
+
+    
+    // MARK: - KVO Rocks!
+    
+    private func startListeningToProperties(viewController: UIViewController) {
+        for key in Properties.all where viewController.respondsToSelector(NSSelectorFromString(key.rawValue)) {
+            viewController.addObserver(self, forKeyPath: key.rawValue, options: [.Initial, .New], context: nil)
+        }
+    }
+    
+    private func stopListeningToProperties(viewController: UIViewController) {
+        for key in Properties.all where viewController.respondsToSelector(NSSelectorFromString(key.rawValue)) {
+            viewController.removeObserver(self, forKeyPath: key.rawValue)
+        }
+    }
+
+    override func observeValueForKeyPath(keyPath: String?, ofObject object: AnyObject?, change: [String : AnyObject]?, context: UnsafeMutablePointer<Void>) {
+        guard let unwrappedKeyPath = keyPath, let property = Properties(rawValue: unwrappedKeyPath) else {
```

Comment:
I was wondering why you'd use an enum for properties instead of constants until I saw this.

👍 👍 Nice touch